Dart-Code/Dart-Code
# Support running/debugging Flutter Driver integration tests

This adds some special treatment for the `test_driver` folder so the files in it can be considered tests, and entry points. This doesn't user `flutter drive` (as it doesn't seem to have a good API mode) but allows you to do the same thing it does by launching the instrumented app and then running tests against it.

- Open `test_driver/app.dart` and press `F5` - this should launch this file (the instrumented version)
- Wait for the app to launch
- Switch to `test_driver.app_test.dart`
- Press `F5` to run the whole file, or use the CodeLens `Run`/`Debug` links to run individual tests

This should give full debugging of both the app and the test, and shouldn't need launch configs. If you do want to be able to launch these from launch configs, you should be able to use basic configs as you'd expect:

```js
 {
      "name": "Integration Tests: Launch App",
      "request": "launch",
      "type": "dart",
      "program": "test_driver/app.dart",
    },
    {
      "name": "Integration Tests: Launch Driver",
      "request": "launch",
      "type": "dart",
      "program": "test_driver/app_test.dart",
    },
```

I made it automatically Hot Restart the app at the start of the test run, as it seemed weird to run with existing state (the [sample tests here](https://flutter.dev/docs/cookbook/testing/integration/introduction) would fail if run multiple times due to the counter state not being reset).
